### PR TITLE
Search other metadata

### LINKED ISSUE #27: Search by other metadata
Search by
* minimum size
* maximum size
* modification time

### PR DESCRIPTION
**Pull request by dannywerner**
add filter screen with function the select min/max date and min/max filesize
filter min/max generated from filelist
fix navigation bug
resolves #27

### CODE REVIEW

**Comment by dannywerner** [comment]
please recheck if the bugfix correct
filterByDate in storagesearch. iam added '*1000'
Date boxDate = new Date(((BoxFile) o).mtime*1000);

File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/FileSearchFilterFragment.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+        if (id == R.id.action_use_filter) {
+            getFragmentManager().popBackStack();
+            mFilterData.mFileSizeMin = (int) (mSbFileSizeMin.getProgress() + mMinFileSize);
+            mFilterData.mFileSizeMax = (int) (mSbFileSizeMax.getProgress() + mMinFileSize);
+            if(mNewMinDate>0)
+            mFilterData.mDateMin = new Date(mNewMinDate * 1000);
+            if(mNewMaxDate>0)
+            mFilterData.mDateMax = new Date(mNewMaxDate * 1000);
```
Comment (audax) [inline qabelbox/src/main/java/de/qabel/qabelbox/fragments/FileSearchFilterFragment.java]: Please always use braces for if-statements.

File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/FilesSearchResultFragment.java [hunk 2]
```diff
@@ -107,7 +115,42 @@ public String getTitle() {
...
+            if (data.mDateMax != null)
+                result.filterByMaximumDate(data.mDateMax);
+
+
+            result.filterByMinimumSize(data.mFileSizeMin);
+            result.filterByMaximumSize(data.mFileSizeMax);
+            fillAdapter(result.getResults());
+            filesAdapter.notifyDataSetChanged();
+        } catch (CloneNotSupportedException e) {
+            e.printStackTrace();
```
Comment (audax) [inline qabelbox/src/main/java/de/qabel/qabelbox/fragments/FilesSearchResultFragment.java]: The Log.e is good enough, no need for a stack trace print

**Comment by dannywerner** [comment]
refactored

**Comment by audax** [comment]
Merged in 9730fd8

**Pull request closed without merging**